controllers/machine_controller_phases.go

```diff
@@ -114,6 +114,19 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, cluster *clus
 		return external.ReconcileOutput{}, err
 	}
 
+	// With the migration from v1alpha2 to v1alpha3, Machine controllers should be the owner for the
+	// infra Machines, hence remove any existing machineset controller owner reference
+	if controller := metav1.GetControllerOf(obj); controller != nil && controller.Kind == "MachineSet" {
+		ownerRefs := util.RemoveOwnerRef(obj.GetOwnerReferences(), metav1.OwnerReference{
+			APIVersion: clusterv1.GroupVersion.Version,
```

Comment:
This should be just `clusterv1.GroupVersion.String()`. 
ApiVersion has both group and version info: `cluster.x-k8s.io/v1alpha3`

Reply:
Makes sense, done